server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/Serializer.scala

```diff
@@ -261,8 +261,28 @@ class Serializer @Inject() (mailboxIdFactory: MailboxId.Factory) {
   private implicit val emailGetRequestReads: Reads[EmailGetRequest] = Json.reads[EmailGetRequest]
   private implicit val emailNotFoundWrites: Writes[EmailNotFound] = Json.valueWrites[EmailNotFound]
   private implicit val messageIdWrites: Writes[MessageId] = id => JsString(id.serialize())
-  private implicit def emailWritesWithPropertyFilter(properties: Properties)(implicit partsWrites: Writes[EmailBodyPart]): Writes[Email] = Json.writes[Email]
-    .transform(obj => properties.filter(obj))
+
+  private implicit def emailWritesWithPropertyFilter(properties: Properties)(implicit partsWrites: Writes[EmailBodyPart]): Writes[Email] = (o: Email) => {
+    properties.filter(Json.obj("id" -> o.id,
+      "size" -> o.size,
+      "bodyStructure" -> o.bodyStructure,
+      "textBody" -> o.textBody,
+      "htmlBody" -> o.htmlBody,
+      "attachments" -> o.attachments,
+      "headers" -> o.headers,
+      "header:MessageId" -> o.headerMessageId,
+      "header:Subject" -> o.headerSubject,
+      "header:From" -> o.headerFrom,
+      "header:To" -> o.headerTo,
+      "header:Cc" -> o.headerCc,
+      "header:Bcc" -> o.headerBcc,
+      "header:ReplyTo" -> o.headerReplyTo,
+      "header:InReplyTo" -> o.headerInReplyTo,
+      "header:Sender" -> o.headerSender,
+      "header:References" -> o.headerReferences,
+      "header:SentAt" -> o.headerSentAt))
```

Comment:
What if I request `header:GaBouZoMeu` ?

(ie: the set of specific header that can be requested is not a finite ensemble, your strategy do not work)

Reply:
I mentioned this problem this morning. May be i didn't make myself clear enough...